src/pokedex_data.c

```diff
@@ -0,0 +1,1235 @@
+        form = ReadBit_Rotom(pokedexData->rotomFormsSeen, formIndex);
+
+        if (form != TERMINAL_4BITS) {
+            numFormsSeen++;
+        } else {
+            break;
+        }
+    }
+
+    return numFormsSeen;
+}
+
+static BOOL FormSeen_Rotom(const PokedexData *pokedexData, u32 species, u8 form)
+{
+    GF_ASSERT(species == SPECIES_ROTOM);
+
+    if (PokedexData_HasSeenSpecies(pokedexData, species) == FALSE) {
+        return FALSE;
+    }
+
+    int formIndex;
+    u32 numFormsSeen = NumFormsSeen_Rotom(pokedexData, species);
+
+    for (formIndex = 0; formIndex < numFormsSeen; formIndex++) {
+        u32 rotomForm = ReadBit_Rotom(pokedexData->rotomFormsSeen, formIndex);
+
+        if (rotomForm == form) {
+            return TRUE;
+        }
+    }
+
+    return FALSE;
+}
+
+static void UpdateForms_Rotom(PokedexData *pokedexData, u32 species, int form)
+{
+    int numFormsSeen;
+
+    GF_ASSERT(species == SPECIES_ROTOM);
+
+    if (FormSeen_Rotom(pokedexData, species, form)) {
+        return;
+    }
+
+    numFormsSeen = NumFormsSeen_Rotom(pokedexData, species);
+
+    if (numFormsSeen < ROTOM_COUNT) {
+        SetBit_Rotom(&pokedexData->rotomFormsSeen, numFormsSeen, form);
+    }
+}
+
+static void UpdateForm(PokedexData *pokedexData, u16 species, Pokemon *pokemon)
+{
+    int form;
+
+    switch (species) {
+    case SPECIES_UNOWN:
+        form = Pokemon_GetForm(pokemon);
+        SetUnownForm(pokedexData, form);
+        break;
+    case SPECIES_BURMY:
+        form = Pokemon_GetValue(pokemon, MON_DATA_FORM, NULL);
+        UpdateForms_ThreeForms(pokedexData, species, form);
+        break;
+    case SPECIES_WORMADAM:
+        form = Pokemon_GetValue(pokemon, MON_DATA_FORM, NULL);
+        UpdateForms_ThreeForms(pokedexData, species, form);
+        break;
+    case SPECIES_SHELLOS:
+        form = Pokemon_GetValue(pokemon, MON_DATA_FORM, NULL);
+        UpdateForms_TwoForms(pokedexData, species, form);
+        break;
+    case SPECIES_GASTRODON:
+        form = Pokemon_GetValue(pokemon, MON_DATA_FORM, NULL);
+        UpdateForms_TwoForms(pokedexData, species, form);
+        break;
+    case SPECIES_DEOXYS:
+        UpdateForms_Deoxys(pokedexData, species, pokemon);
+        break;
+    case SPECIES_SHAYMIN:
+        form = Pokemon_GetValue(pokemon, MON_DATA_FORM, NULL);
+        UpdateForms_TwoForms(pokedexData, species, form);
+        break;
+    case SPECIES_GIRATINA:
+        form = Pokemon_GetValue(pokemon, MON_DATA_FORM, NULL);
+        UpdateForms_TwoForms(pokedexData, species, form);
+        break;
+    case SPECIES_ROTOM:
+        form = Pokemon_GetValue(pokemon, MON_DATA_FORM, NULL);
+        UpdateForms_Rotom(pokedexData, species, form);
+        break;
+    default:
+        break;
+    }
+}
+
+static void UpdateLanguage(PokedexData *pokedexData, u16 species, u32 language)
+{
+    int bitIndex = species;
+    int languageIndex = sub_020986CC(language);
+
+    if (languageIndex == 6) {
```

Comment:
**question**: Should we mark this as an unused constant in `include/global/pm_version.h`?

Reply:
Strangely, it appears that language 2 is set to 6 in #323. I am not 100% sure, but based on the dex entries it looks like 2==diamond and 6==platinum, at least for the dex. Perhaps a holdover from Pearl?

Reply:
Interesting... I would think that 2 would be Platinum, and 6 would be Pearl, given the context that most of the version-specific code checks for `DIAMOND || PLATINUM`, treating `PEARL` as a separate entity. Can you confirm?

Reply:
It would likely be more clear in this specific instance if `include/global/pm_version.h` had a value for this unused language. Would that be within the scope of this PR?